tests/reg_issue1069/reg_issue1069.fz

```diff
@@ -0,0 +1,36 @@
+# This file is part of the Fuzion language implementation.
+#
+# The Fuzion language implementation is free software: you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as published
+# by the Free Software Foundation, version 3 of the License.
+#
+# The Fuzion language implementation is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU General Public License along with The
+# Fuzion language implementation.  If not, see <https://www.gnu.org/licenses/>.
+
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion test
+#
+# -----------------------------------------------------------------------
+
+reg_issue1069 is
+
+  a := [0.3, option 3.142, nil, 42]
+  for v in a do say "$v {type_of v}"
+
+  f(b bool) =>
+    if b
+      42
+    else
+      option 3.14
+
+  say (f true)
+  say (f false)
```

Comment:
Would be good to also do `say (type_of (f true))` and  `say (type_of (f false))` (even though this is the same)